tests/baselines/reference/nullishCoalescingOperator1.js

```diff
@@ -0,0 +1,60 @@
+//// [nullishCoalescingOperator1.ts]
+declare const a1: string | undefined | null
+declare const a2: string | undefined | null
+declare const a3: string | undefined | null
+declare const a4: string | undefined | null
+
+declare const b1: number | undefined | null
+declare const b2: number | undefined | null
+declare const b3: number | undefined | null
+declare const b4: number | undefined | null
+
+declare const c1: boolean | undefined | null
+declare const c2: boolean | undefined | null
+declare const c3: boolean | undefined | null
+declare const c4: boolean | undefined | null
+
+interface I { a: string }
+declare const d1: I | undefined | null
+declare const d2: I | undefined | null
+declare const d3: I | undefined | null
+declare const d4: I | undefined | null
+
+const aa1 = a1 ?? 'whatever';
+const aa2 = a2 ?? 'whatever';
+const aa3 = a3 ?? 'whatever';
+const aa4 = a4 ?? 'whatever';
+
+const bb1 = b1 ?? 1;
+const bb2 = b2 ?? 1;
+const bb3 = b3 ?? 1;
+const bb4 = b4 ?? 1;
+
+const cc1 = c1 ?? true;
+const cc2 = c2 ?? true;
+const cc3 = c3 ?? true;
+const cc4 = c4 ?? true;
+
+const dd1 = d1 ?? {b: 1};
+const dd2 = d2 ?? {b: 1};
+const dd3 = d3 ?? {b: 1};
+const dd4 = d4 ?? {b: 1};
+
+//// [nullishCoalescingOperator1.js]
+"use strict";
+var aa1 = typeof a1 !== "undefined" && a1 !== null ? a1 : 'whatever';

```

Comment:
I think this condition is reversed. 😄 

You either want equality or to reverse the branches. I'd prefer just switching to `===`/`==`.

Reply:
Just as a heads up, we might want to just go with `a1 == null`, since `document.all` is a weird anomaly of the language anyway that should ideally never be used in a `??` expression.